programs/fileio.c

```diff
@@ -733,20 +734,15 @@ static size_t FIO_createDictBuffer(void** bufferPtr, const char* fileName, FIO_p
 
 #if (PLATFORM_POSIX_VERSION > 0)
 #include <sys/mman.h>
-static void* FIO_mmap(size_t fileSize, int fileHandle)
+static int FIO_munmap(const FIO_Dict_t* dict)
```

Comment:
It feels strange to use `const` here,
because the action is not just a simple "read",
it is effectively modifying the underlying state, 
therefore I would prefer `FIO_Dict_t* dict` type.

You could use this opportunity to set `dictBuffer=NULL` and `dictBufferSize=0` after the `munmap()` operation,
thus ensuring the `unmap` operation is properly tracked and there is no dangling pointer left behind.